everest

```diff
@@ -558,39 +558,38 @@ OCAML
   echo "... nuget found in PATH"
 
   # We need Scons for Vale which needs Python 2.7
-  #
-  # What we really should do instead of checking for a directory name is looking at the
-  # registry. These are the four possible values:
-  #
-  # regtool get \HKLM\Software\Python\PythonCore\2.7\InstallPath\
-  # regtool get \HKCU\Software\Python\PythonCore\2.7\InstallPath\
-  # regtool get \HKLM\Software\WOW6432Node\Python\PythonCore\2.7\InstallPath\
-  # regtool get \HKCU\Software\WOW6432Node\Python\PythonCore\2.7\InstallPath\
-  #
-  #
   if is_windows; then
-    if ! [ -d "/cygdrive/c/Python27" ]; then
-      red "ERROR: C:\\Python27 not found -- install a *WINDOWS* Python 2.7? (y/n)"
+    if [ -z ${PY27DIR+x} ]; then
+      PY27DIR=$(regtool get /HKLM/Software/Python/PythonCore/2.7/InstallPath/)
+      if ! [ -d $PY27DIR ]; then 
+          PY27DIR=$(regtool get /HKCU/Software/Python/PythonCore/2.7/InstallPath/)
+      fi
+      if ! [ -d $PY27DIR ]; then
+          PY27DIR="/cygdrive/c/Python27"
+      fi
+    fi
+    if ! [ -d $PY27DIR ]; then
+      red "ERROR: Python not found -- install a *WINDOWS* Python 2.7? (y/n)"
       if prompt_yes true "exit 1"; then
         wget $PYTHON_INSTALL_POINT
         chmod a+x python-$PYTHON_VER.amd64.msi
         msiexec /i python-$PYTHON_VER.amd64.msi /quiet
-        windows_check_or_modify_env_dest_file "python.exe" "C:\\Python27"
+        windows_check_or_modify_env_dest_file "python.exe" "$PY27DIR"
```

Comment:
there is a change of behavior: with your patch, windows_check_or_modify_env_dest_file may now be passed a cygwin path instead of a windows path; is this correct?

Reply:
Yes, I think so. I'm struggling with path problems all over the place, I'll fix that once I figured out what the right thing to do is.